backend/controllers/globalSearch.js

```diff
@@ -1,56 +1,131 @@
+} = require('../config')
+
+module.exports = ({ paginationService, offenderSearchApi, oauthApi, logError }) => {
+  const searchByOffender = (context, offenderNo, gender, location, dateOfBirth, pageLimit) =>
+    offenderSearchApi.globalSearch(
+      context,
+      {
+        prisonerIdentifier: offenderNo,
+        gender,
+        location,
+        dateOfBirth,
+        includeAliases: true,
+      },
+      pageLimit
+    )
+
+  const searchByName = (context, name, gender, location, dateOfBirth, pageLimit) => {
     const [lastName, firstName] = name.split(' ')
-    return globalSearchApi.globalSearch(context, {
-      lastName,
-      firstName,
-      gender,
-      location,
-      dateOfBirth,
-      includeAliases: true,
-    })
+    return offenderSearchApi.globalSearch(
+      context,
+      {
+        lastName,
+        firstName,
+        gender,
+        location,
+        dateOfBirth,
+        includeAliases: true,
+      },
+      pageLimit
+    )
+  }
+
+  const backWhiteList = { licences: licencesUrl }
+
+  const indexPage = (req, res) => {
+    const { referrer } = req.query
+    return res.render('globalSearch/globalSearch.njk', { backLink: backWhiteList[referrer], dpsUrl, referrer })
   }
 
-  const globalSearch = async (context, searchText, gender, location, dateOfBirth) => {
-    log.info(`In globalSearch, searchText=${searchText}`)
-    if (!searchText) {
-      return []
+  const resultsPage = async (req, res) => {
+    let prisonerResults = []
+    const { searchText, pageLimitOption, pageOffsetOption, referrer, ...filters } = req.query
+    const { genderFilter, locationFilter, dobDay, dobMonth, dobYear } = filters
+    const pageLimit = (pageLimitOption && parseInt(pageLimitOption, 10)) || 20
+    const pageOffset = (pageOffsetOption && parseInt(pageOffsetOption, 10)) || 0
+
+    const { dobErrors, dobIsValid, dateOfBirth } = dobValidation(dobDay, dobMonth, dobYear)
+    const errors = [...dobErrors]
+
+    const context = {
+      ...res.locals,
+      requestHeaders: {
+        'page-offset': pageOffset,
+        'page-limit': pageLimit,
+      },
     }
-    const text = searchText
-      .replace(/,/g, ' ')
-      .replace(/\s\s+/g, ' ')
-      .trim()
-    const data = await (isPrisonerIdentifier(text)
-      ? searchByOffender(context, text, gender, location, dateOfBirth)
-      : searchByName(context, text, gender, location, dateOfBirth))
-    log.info(data.length, 'globalSearch data received')
-
-    return data.map(offender => {
-      const formattedDateOfBirth =
-        offender.dateOfBirth && moment(offender.dateOfBirth, 'YYYY-MM-DD').format('DD/MM/YYYY')
-      const uiId = shortid.generate()
-
-      return {
-        ...offender,
-        dateOfBirth: formattedDateOfBirth,
-        uiId,
+
+    try {
+      if (searchText && !errors.length) {
+        log.info(`In globalSearch, searchText=${searchText}`)
+
+        const dateOfBirthFilter = dobIsValid ? dateOfBirth.format('YYYY-MM-DD') : undefined
```

Comment:
Would the nullish operator work here? ` dobIsValid ?? dateOfBirth.format('YYYY-MM-DD')`

Reply:
Unfortunately not, with the way we then call the offenderSearchApi endpoint.